### ISSUE TITLE
Default login information failing

### ISSUE THREAD

**Issue by enigma-x** [byceps/byceps#13]
I followed the documentation to install byceps using docker-compose.

Despite the documentation is somewhat incomplete (where to obtain the compose.yaml to build the containers in the first place is quite unclear), I managed to get it running.

Both admin and demo sites appear to be running fine, however, the default login information provided in the documentation (DemoAdmin/demodemo) is not working.

**Comment by enigma-x** [comment]
In the meantime I figured out that the credentials are actually right and accepted, but I do not get past the login screen. Doesn't appear to be browser related, tested with both Firefox 118.02 and Chrome 118.0.5993.88.

(Tested by doing a negative test by typing a different password actually triggers the expected error message).

**Comment by enigma-x** [comment]
```
$ docker-compose up
byceps_redis_1 is up-to-date
byceps_db_1 is up-to-date
Starting byceps_byceps-worker_1 ... done
Starting byceps_byceps-site_1   ... done
Starting byceps_byceps-admin_1  ... done
Starting byceps_web_1           ... done
Attaching to byceps_redis_1, byceps_db_1, byceps_byceps-admin_1, byceps_byceps-site_1, byceps_byceps-worker_1, byceps_web_1
byceps-admin_1   | *** Starting uWSGI 2.0.22 (64bit) on [Tue Oct 24 13:25:24 2023] ***
byceps-admin_1   | compiled with version: 12.2.0 on 24 October 2023 12:35:25
byceps-admin_1   | os: Linux-6.1.0-13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29)
byceps-admin_1   | nodename: f07b6bb7239d
byceps-admin_1   | machine: x86_64
byceps-admin_1   | clock source: unix
byceps-admin_1   | detected number of CPU cores: 2
byceps-admin_1   | current working directory: /home/byceps
byceps-admin_1   | detected binary path: /home/byceps/.local/bin/uwsgi
byceps-admin_1   | !!! no internal routing support, rebuild with pcre support !!!
byceps-admin_1   | *** WARNING: you are running uWSGI without its master process manager ***
byceps-admin_1   | your memory page size is 4096 bytes
byceps-admin_1   | detected max file descriptor number: 1048576
byceps-admin_1   | lock engine: pthread robust mutexes
byceps-admin_1   | thunder lock: disabled (you can enable it with --thunder-lock)
byceps-admin_1   | uwsgi socket 0 bound to TCP address 0.0.0.0:8080 fd 3
byceps-admin_1   | uwsgi socket 1 bound to TCP address 0.0.0.0:5000 fd 4
byceps-admin_1   | Python version: 3.11.6 (main, Oct 12 2023, 10:04:56) [GCC 12.2.0]
byceps-admin_1   | Python main interpreter initialized at 0x7f7fc9b5ceb8
byceps-admin_1   | python threads support enabled
byceps-admin_1   | your server socket listen backlog is limited to 100 connections
byceps-admin_1   | your mercy for graceful operations on workers is 60 seconds
byceps-admin_1   | mapped 72920 bytes (71 KB) for 1 cores
byceps-admin_1   | *** Operational MODE: single process ***
byceps-admin_1   | *** uWSGI is running in multiple interpreter mode ***
byceps-admin_1   | spawned uWSGI worker 1 (and the only) (pid: 1, cores: 1)
byceps-site_1    | *** Starting uWSGI 2.0.22 (64bit) on [Tue Oct 24 13:25:24 2023] ***
byceps-site_1    | compiled with version: 12.2.0 on 24 October 2023 12:35:25
byceps-site_1    | os: Linux-6.1.0-13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29)
byceps-site_1    | nodename: 3ad055b15a86
byceps-site_1    | machine: x86_64
byceps-site_1    | clock source: unix
byceps-site_1    | detected number of CPU cores: 2
byceps-site_1    | current working directory: /home/byceps
byceps-site_1    | detected binary path: /home/byceps/.local/bin/uwsgi
byceps-site_1    | !!! no internal routing support, rebuild with pcre support !!!
byceps-site_1    | *** WARNING: you are running uWSGI without its master process manager ***
byceps-site_1    | your memory page size is 4096 bytes
byceps-site_1    | detected max file descriptor number: 1048576
byceps-site_1    | lock engine: pthread robust mutexes
byceps-site_1    | thunder lock: disabled (you can enable it with --thunder-lock)
byceps-site_1    | uwsgi socket 0 bound to TCP address 0.0.0.0:8080 fd 3
byceps-site_1    | uwsgi socket 1 bound to TCP address 0.0.0.0:5000 fd 4
byceps-site_1    | Python version: 3.11.6 (main, Oct 12 2023, 10:04:56) [GCC 12.2.0]
byceps-site_1    | Python main interpreter initialized at 0x7f4307cafeb8
byceps-site_1    | python threads support enabled
byceps-site_1    | your server socket listen backlog is limited to 100 connections
byceps-site_1    | your mercy for graceful operations on workers is 60 seconds
byceps-site_1    | mapped 72920 bytes (71 KB) for 1 cores
byceps-site_1    | *** Operational MODE: single process ***
byceps-site_1    | *** uWSGI is running in multiple interpreter mode ***
byceps-site_1    | spawned uWSGI worker 1 (and the only) (pid: 1, cores: 1)
web_1            | /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
web_1            | /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
web_1            | /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
web_1            | 10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
db_1             | The files belonging to this database system will be owned by user "postgres".
db_1             | This user must also own the server process.
db_1             |
db_1             | The database cluster will be initialized with locale "en_US.utf8".
db_1             | The default database encoding has accordingly been set to "UTF8".
db_1             | The default text search configuration will be set to "english".
db_1             |
db_1             | Data page checksums are disabled.
db_1             |
db_1             | fixing permissions on existing directory /var/lib/postgresql/data ... ok
db_1             | creating subdirectories ... ok
db_1             | selecting dynamic shared memory implementation ... posix
db_1             | selecting default max_connections ... 100
db_1             | selecting default shared_buffers ... 128MB
db_1             | selecting default time zone ... Etc/UTC
db_1             | creating configuration files ... ok
db_1             | running bootstrap script ... ok
db_1             | performing post-bootstrap initialization ... ok
db_1             | syncing data to disk ... ok
db_1             |
db_1             | initdb: warning: enabling "trust" authentication for local connections
db_1             | initdb: hint: You can change this by editing pg_hba.conf or using the option -A, or --auth-local and --auth-host, the next time you run initdb.
db_1             |
db_1             | Success. You can now start the database server using:
db_1             |
db_1             |     pg_ctl -D /var/lib/postgresql/data -l logfile start
db_1             |
db_1             | waiting for server to start....2023-10-24 13:24:56.804 UTC [48] LOG:  starting PostgreSQL 15.4 (Debian 15.4-2.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
db_1             | 2023-10-24 13:24:56.807 UTC [48] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1             | 2023-10-24 13:24:56.816 UTC [51] LOG:  database system was shut down at 2023-10-24 13:24:56 UTC
db_1             | 2023-10-24 13:24:56.821 UTC [48] LOG:  database system is ready to accept connections
db_1             |  done
db_1             | server started
db_1             | CREATE DATABASE
db_1             |
db_1             |
db_1             | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
db_1             |
db_1             | 2023-10-24 13:24:57.008 UTC [48] LOG:  received fast shutdown request
db_1             | waiting for server to shut down....2023-10-24 13:24:57.010 UTC [48] LOG:  aborting any active transactions
db_1             | 2023-10-24 13:24:57.013 UTC [48] LOG:  background worker "logical replication launcher" (PID 54) exited with exit code 1
db_1             | 2023-10-24 13:24:57.014 UTC [49] LOG:  shutting down
db_1             | 2023-10-24 13:24:57.016 UTC [49] LOG:  checkpoint starting: shutdown immediate
db_1             | 2023-10-24 13:24:57.064 UTC [49] LOG:  checkpoint complete: wrote 918 buffers (5.6%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.024 s, sync=0.016 s, total=0.050 s; sync files=301, longest=0.004 s, average=0.001 s; distance=4223 kB, estimate=4223 kB
db_1             | 2023-10-24 13:24:57.069 UTC [48] LOG:  database system is shut down
db_1             |  done
db_1             | server stopped
db_1             |
db_1             | PostgreSQL init process complete; ready for start up.
db_1             |
db_1             | 2023-10-24 13:24:57.132 UTC [1] LOG:  starting PostgreSQL 15.4 (Debian 15.4-2.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
db_1             | 2023-10-24 13:24:57.132 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db_1             | 2023-10-24 13:24:57.132 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db_1             | 2023-10-24 13:24:57.137 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1             | 2023-10-24 13:24:57.144 UTC [64] LOG:  database system was shut down at 2023-10-24 13:24:57 UTC
db_1             | 2023-10-24 13:24:57.149 UTC [1] LOG:  database system is ready to accept connections
redis_1          | 1:C 24 Oct 2023 13:24:56.016 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis_1          | 1:C 24 Oct 2023 13:24:56.016 * oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis_1          | 1:C 24 Oct 2023 13:24:56.016 * Redis version=7.2.2, bits=64, commit=00000000, modified=0, pid=1, just started
redis_1          | 1:C 24 Oct 2023 13:24:56.016 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
redis_1          | 1:M 24 Oct 2023 13:24:56.016 * monotonic clock: POSIX clock_gettime
redis_1          | 1:M 24 Oct 2023 13:24:56.016 * Running mode=standalone, port=6379.
redis_1          | 1:M 24 Oct 2023 13:24:56.016 * Server initialized
redis_1          | 1:M 24 Oct 2023 13:24:56.017 * Ready to accept connections tcp
web_1            | 10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
web_1            | /docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
web_1            | /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
web_1            | 20-envsubst-on-templates.sh: Running envsubst on /etc/nginx/templates/default.conf.template to /etc/nginx/conf.d/default.conf
web_1            | /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
web_1            | /docker-entrypoint.sh: Configuration complete; ready for start up
web_1            | 2023/10/24 13:25:25 [notice] 1#1: using the "epoll" event method
web_1            | 2023/10/24 13:25:25 [notice] 1#1: nginx/1.25.2
web_1            | 2023/10/24 13:25:25 [notice] 1#1: built by gcc 12.2.0 (Debian 12.2.0-14)
web_1            | 2023/10/24 13:25:25 [notice] 1#1: OS: Linux 6.1.0-13-amd64
web_1            | 2023/10/24 13:25:25 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
web_1            | 2023/10/24 13:25:25 [notice] 1#1: start worker processes
web_1            | 2023/10/24 13:25:25 [notice] 1#1: start worker process 35
web_1            | 2023/10/24 13:25:25 [notice] 1#1: start worker process 36
byceps-admin_1   | 2023-10-24 13:25:26 [info     ] Sentry integration: disabled (no DSN configured)
byceps-site_1    | 2023-10-24 13:25:26 [info     ] Sentry integration: disabled (no DSN configured)
byceps-worker_1  | 2023-10-24 13:25:27 [info     ] Sentry integration: disabled (no DSN configured)
byceps-worker_1  | 13:25:27 Worker rq:worker:963d02496d5941f8a3ebb564d1ec46dc started with PID 1, version 1.15.1
byceps-worker_1  | 13:25:27 Subscribing to channel rq:pubsub:963d02496d5941f8a3ebb564d1ec46dc
byceps-worker_1  | 13:25:27 *** Listening on default...
byceps-worker_1  | 13:25:27 Scheduler for default started with PID 8
byceps-worker_1  | 13:25:27 Cleaning registries for queue: default
byceps-site_1    | 2023-10-24 13:25:27 [info     ] Application created            app_mode=site debug=disabled debug_toolbar=disabled style_guide=disabled
byceps-site_1    | WSGI app 0 (mountpoint='') ready in 3 seconds on interpreter 0x7f4307cafeb8 pid: 1 (default app)
byceps-admin_1   | 2023-10-24 13:25:27 [info     ] Application created            app_mode=admin debug=disabled debug_toolbar=disabled metrics=disabled rq_dashboard=enabled style_guide=disabled
byceps-admin_1   | WSGI app 0 (mountpoint='') ready in 3 seconds on interpreter 0x7f7fc9b5ceb8 pid: 1 (default app)
byceps-admin_1   | [pid: 1|app: 0|req: 1/1] 172.22.0.2 () {44 vars in 729 bytes} [Tue Oct 24 13:25:31 2023] GET /authentication/log_in => generated 2641 bytes in 32 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 0)
web_1            | 172.22.0.2 - - [24/Oct/2023:13:25:31 +0000] "GET /authentication/log_in HTTP/1.1" 200 2641 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0" "-"
web_1            | 172.22.0.2 - - [24/Oct/2023:13:25:32 +0000] "GET /static/style/common.css HTTP/1.1" 304 0 "http://10.52.30.21:8081/authentication/log_in" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0" "-"
web_1            | 172.22.0.2 - - [24/Oct/2023:13:25:32 +0000] "GET /static/style/admin.css HTTP/1.1" 304 0 "http://10.52.30.21:8081/authentication/log_in" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0" "-"
web_1            | 172.22.0.2 - - [24/Oct/2023:13:25:32 +0000] "GET /static/behavior/common.js HTTP/1.1" 304 0 "http://10.52.30.21:8081/authentication/log_in" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0" "-"
byceps-admin_1   | 2023-10-24 13:25:37 [info     ] User logged in                 scope=admin screen_name=DemoAdmin user_id=27ce732d-2c6f-41e6-b71b-406336d69036
byceps-admin_1   | [pid: 1|app: 0|req: 2/2] 172.22.0.2 () {52 vars in 943 bytes} [Tue Oct 24 13:25:37 2023] POST /authentication/log_in => generated 189 bytes in 249 msecs (HTTP/1.1 302) 5 headers in 482 bytes (1 switches on core 0)
web_1            | 172.22.0.2 - - [24/Oct/2023:13:25:37 +0000] "POST /authentication/log_in HTTP/1.1" 302 189 "http://10.52.30.21:8081/authentication/log_in" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0" "-"
web_1            | 172.22.0.2 - - [24/Oct/2023:13:25:37 +0000] "GET / HTTP/1.1" 307 231 "http://10.52.30.21:8081/authentication/log_in" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0" "-"
byceps-admin_1   | [pid: 1|app: 0|req: 3/3] 172.22.0.2 () {46 vars in 748 bytes} [Tue Oct 24 13:25:37 2023] GET / => generated 231 bytes in 0 msecs (HTTP/1.1 307) 4 headers in 144 bytes (1 switches on core 0)
byceps-admin_1   | [pid: 1|app: 0|req: 4/4] 172.22.0.2 () {46 vars in 790 bytes} [Tue Oct 24 13:25:37 2023] GET /authentication/log_in => generated 2641 bytes in 1 msecs (HTTP/1.1 200) 3 headers in 95 bytes (1 switches on core 0)
web_1            | 172.22.0.2 - - [24/Oct/2023:13:25:37 +0000] "GET /authentication/log_in HTTP/1.1" 200 2641 "http://10.52.30.21:8081/authentication/log_in" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0" "-"
`

**Comment by homeworkprod** [comment]
Hey @enigma-x, thanks for checking out BYCEPS! Sorry you ran into trouble.

Could it be that you are trying to access the apps with unencrypted HTTP? In that case a security feature prevents the cookie from being set.

You can set [`SESSION_COOKIE_SECURE`](https://byceps.readthedocs.io/en/latest/config/index.html#SESSION_COOKIE_SECURE) to `false` to disable this behavior.

In the Compose file, adding it like this should work:
```yaml
x-byceps-base-env: &byceps-base-env
    # [… other environment variables here …]
    SESSION_COOKIE_SECURE: "false"
```

**Comment by enigma-x** [comment]
Yes, this was the case indeed. Thanks for clearing this up.

**Comment by homeworkprod** [comment]
> Despite the documentation is somewhat incomplete (where to obtain the compose.yaml to build the containers in the first place is quite unclear), I managed to get it running.

I have clarified that there is a Compose file included in the repository in 4ae8e4c669818c67d936de2ac0e6212f687a5237.

Regarding the need for TLS to use secure session cookies and the option to disable them, I have extended the documentation in bf330cfe9225a7bc2b0e754f1fcc8a52f35d96fe.
